sdk/core/core-crypto/src/utils/base64.browser.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/// <reference lib="dom" />
+
+/**
+ * Converts a base64 string into a byte array.
+ * @param content - The base64 string to convert.
+ * @hidden
+ * @internal
+ */
+export function base64ToBytes(content: string): Uint8Array {
+  if (typeof atob !== "function") {
+    throw new Error(`Your browser environment is missing the global "atob" function.`);
+  }
+
+  const binary = atob(content);
+  const bytes = new Uint8Array(binary.length);
+  for (let i = 0; i < binary.length; i++) {
+    bytes[i] = binary.charCodeAt(i);
+  }
+
+  return bytes;
+}
+
+/**
+ * Converts an ArrayBuffer to base64 string.
+ * @param buffer - Raw binary data.
+ * @hidden
+ * @internal
+ */
+export function bufferToBase64(buffer: ArrayBuffer): string {
```

Comment:
do we need to provide these helpers publicly too? I'm pretty sure there are some credential types out there that need base64 support

Reply:
Yes please!

Reply:
Also, perhaps consider using this `buffer` library: https://www.npmjs.com/package/buffer

I am using it in attestation tests: 
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/attestation/attestation/test/utils/Buffer.browser.ts
https://github.com/Azure/azure-sdk-for-js/blob/53554277be67677adba188eedea6f164f7c1fc35/sdk/attestation/attestation/test/utils/decodeJWT.ts#L47

Reply:
I was debating exposing these helpers. Would we need these outside of crypto? e.g. do we need like a core-encoder package?

I thought it was interesting that every usage of the sha256 hmac function today passed the key as a base64 string already...does this feel acceptable to continue doing this here or should I accept something like a byte array/buffer?

> Also, perhaps consider using this buffer

I did consider this. I actually wanted to stay away from Buffer since it can be a fairly large package and increases browser bundle sizes. Also I'd expect the built in atob/btoa functions in browsers to perform better than a JS Buffer implementation. Now, I think we depend on it in core-amqp and core-http but wasn't sure if we were going to continue that with corev2. If we are then it won't be a big deal to include it in this package.

Reply:
Thanks Chris. Perhaps we can discuss in our next weekly meeting whether having a `core-utils/core-encoding` is a good idea. I am personally onboard with that.

Reply:
I'd like to avoid exporting these helpers until we know we need them to avoid adding to the public API prematurely. I do think a core-encoding package may be a better place for these to live long-term.